src/vehicle.cpp

```diff
@@ -2892,7 +2892,7 @@ int vehicle::basic_consumption( const itype_id &ftype ) const
 int vehicle::consumption_per_hour( const itype_id &ftype, int fuel_rate_w ) const
 {
     item fuel = item( ftype );
-    if( fuel_rate_w == 0 || fuel.has_flag( "PERPETUAL" ) ) {
+    if( fuel_rate_w == 0 || fuel.has_flag( "PERPETUAL" ) || !( engine_on ) ) {
```

Comment:
remove the parentheses from engine_on.  it's a bool, negating it is a normal and uncomplicated thing to do.

Thanks for finding and debugging this.

Reply:
Sorry, I got used to stuff like that while working on java stuff. Which is *also* alien to me. Change made.